tests/providers/google/cloud/utils/test_field_sanitizer.py

```diff
@@ -0,0 +1,233 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import unittest
+
+from airflow.providers.google.cloud.utils.field_sanitizer import GcpBodyFieldSanitizer
+
+
+class TestGcpBodyFieldSanitizer(unittest.TestCase):
+    def test_sanitize_should_sanitize_empty_body_and_fields(self):
+        body = {}
+        fields = []
+
+        sanitizer = GcpBodyFieldSanitizer(fields)
+        sanitizer.sanitize(body)
+
+        self.assertEqual({}, body)
+
+    def test_sanitize_should_not_fail_with_none_body(self):
+        body = None
+        fields = []
+
+        sanitizer = GcpBodyFieldSanitizer(fields)
+        sanitizer.sanitize(body)
+
+        self.assertIsNone(body)
+
+    def test_sanitize_should_fail_with_none_fields(self):
+        body = {}
+        fields = None
+
+        sanitizer = GcpBodyFieldSanitizer(fields)
+
+        with self.assertRaises(TypeError):
+            sanitizer.sanitize(body)
+
+    def test_sanitize_should_not_fail_if_field_is_absent_in_body(self):
+        body = {}
+        fields = ["kind"]
+
+        sanitizer = GcpBodyFieldSanitizer(fields)
+        sanitizer.sanitize(body)
+
+        self.assertEqual({}, body)
+
+    def test_sanitize_should_not_remove_fields_for_incorrect_specification(self):
+        actual_body = [
+            {"kind": "compute#instanceTemplate", "name": "instance"},
+            {"kind": "compute#instanceTemplate1", "name": "instance1"},
+            {"kind": "compute#instanceTemplate2", "name": "instance2"},
+        ]
+        body = actual_body
```

Comment:
```suggestion
        body = deepcopy(actual_body)
```

Reply:
correc 👍  dicts share the reference, hence require the deepcopy.

Reply:
Can you please look at this? https://github.com/apache/airflow/issues/8278#issuecomment-663153829 @mik-laj